Sources/Common/Transform/Transform/index.js

```diff
@@ -92,6 +94,279 @@ function vtkTransform(publicAPI, model) {
+    const inArr = inVectors.getData();
+    const outArr = outVectors.getData();
+    const tmp = [0, 0, 0];
+    for (let i = 0; i < inArr.length; i += 3) {
+      tmp[0] = inArr[i];
+      tmp[1] = inArr[i + 1];
+      tmp[2] = inArr[i + 2];
+      publicAPI.transformVector(tmp, tmp);
+      vtkMath.normalize(tmp);
+      outArr[i] = tmp[0];
+      outArr[i + 1] = tmp[1];
+      outArr[i + 2] = tmp[2];
+    }
+  };
+
+  /**
+   * Transform points, normals, and vectors simultaneously.
+   * @param {vtkPoints} inPts Input points
+   * @param {vtkPoints} outPts Output points
+   * @param {vtkDataArray} inNormals Input normals
+   * @param {vtkDataArray} outNormals Output normals
+   * @param {vtkDataArray} inVectors Input vectors
+   * @param {vtkDataArray} outVectors Output vectors
+   * @param {Array<vtkDataArray>} inVectorsArr Optional input vector arrays
+   * @param {Array<vtkDataArray>} outVectorsArr Optional output vector arrays
+   */
+  publicAPI.transformPointsNormalsVectors = (
+    inPts,
+    outPts,
+    inNormals,
+    outNormals,
+    inVectors,
+    outVectors,
+    inVectorsArr = null,
+    outVectorsArr = null
+  ) => {
+    const n = inPts.getNumberOfPoints();
+    const point = new Float64Array(3);
+    const nOptionalVectors = inVectorsArr ? inVectorsArr.length : 0;
+
+    for (let ptId = 0; ptId < n; ptId++) {
+      // Transform point
+      inPts.getPoint(ptId, point);
+      publicAPI.transformPoint(point, point);
+      outPts.setPoint(ptId, ...point);
+
+      // Transform vectors
+      if (inVectors) {
+        const inData = inVectors.getData();
+        const outData = outVectors.getData();
+        point[0] = inData[ptId * 3];
+        point[1] = inData[ptId * 3 + 1];
+        point[2] = inData[ptId * 3 + 2];
+        publicAPI.transformVector(point, point);
+        outData[ptId * 3] = point[0];
+        outData[ptId * 3 + 1] = point[1];
+        outData[ptId * 3 + 2] = point[2];
+      }
+
+      // Transform optional vectors
+      if (inVectorsArr) {
+        for (let iArr = 0; iArr < nOptionalVectors; iArr++) {
+          const inData = inVectorsArr[iArr].getData();
+          const outData = outVectorsArr[iArr].getData();
+          point[0] = inData[ptId * 3];
+          point[1] = inData[ptId * 3 + 1];
+          point[2] = inData[ptId * 3 + 2];
+          publicAPI.transformVector(point, point);
+          outData[ptId * 3] = point[0];
+          outData[ptId * 3 + 1] = point[1];
+          outData[ptId * 3 + 2] = point[2];
+        }
+      }
+
+      // Transform normals
+      if (inNormals) {
+        const inData = inNormals.getData();
+        const outData = outNormals.getData();
+        point[0] = inData[ptId * 3];
+        point[1] = inData[ptId * 3 + 1];
+        point[2] = inData[ptId * 3 + 2];
+        publicAPI.transformNormal(point, point);
+        outData[ptId * 3] = point[0];
+        outData[ptId * 3 + 1] = point[1];
+        outData[ptId * 3 + 2] = point[2];
+      }
+    }
+  };
```

Comment:
you could call `modified()` on `outPts`, `outNormals`, `outVectors` and `outVectorsArr[x]` if at least one point,normal,vector got modified.

Reply:
You mean something like this ?

```diff
-      inPts.getPoint(ptId, point);
+      inPoints.getPoint(ptId, point);
       publicAPI.transformPoint(point, point);
-      outPts.setPoint(ptId, ...point);
+      outPoints.setPoint(ptId, ...point);
+      modified = true;
 
       // Transform vectors
       if (inVectors) {
@@ -366,6 +368,19 @@ function vtkTransform(publicAPI, model) {
         outData[ptId * 3 + 2] = point[2];
       }
     }
+
+    if (modified) {
+      outPoints.modified();
+      if (inVectors) {
+        outVectors.modified();
+      }
+      if (inNormals) {
+        outNormals.modified();
+      }
+      if (inVectorsArr) {
+        outVectorsArr.forEach((arr) => arr.modified());
+      }
+    }
```

Reply:
@finetjul can you check the proposed diff in the previous comment ?